File: src/main/java/io/github/dqualizer/dqlang/types/adapter/constants/LoadTestConstants.java
package io.github.dqualizer.dqlang.types.adapter.constants;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.adapter.accuracy.Accuracy;
import io.github.dqualizer.dqlang.types.adapter.symbolics.Symbolics;
import lombok.Getter;
import lombok.ToString;

/**
 * Java class for load test constants definitions
 */
@Getter
@ToString
public class LoadTestConstants {

    private Accuracy accuracy;
    @JsonProperty("technical_constants")
    private TechnicalConstants technicalConstants;
    private Symbolics symbolics;
    @JsonProperty("response_time")
    private ResponseTime responseTime;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/constants/loadTest/LoadTestConstants.kt
package io.github.dqualizer.dqlang.types.adapter.constants.loadTest

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.adapter.accuracy.Accuracy
import io.github.dqualizer.dqlang.types.adapter.constants.SatisfactionValue
import io.github.dqualizer.dqlang.types.adapter.symbolics.Symbolics

/**
 * Java class for load test constants definitions
 */
data class LoadTestConstants(
  val accuracy: Accuracy,

  @JsonProperty("technical_constants")
  val technicalConstants: TechnicalConstants,

  val symbolics: Symbolics,

  @JsonProperty("response_time")
  val responseTime: SatisfactionValue
)
